mirgecom/simutil.py

```diff
@@ -104,46 +104,94 @@ def exact_sim_checkpoint(discr, exact_soln, visualizer, eos, q,
     if do_viz is False and do_status is False:
         return 0
 
-    actx = q[0].array_context
-    nodes = thaw(actx, discr.nodes())
-    rank = 0
-
-    if comm is not None:
-        rank = comm.Get_rank()
+    if profiler is not None:
+        profiler.StartTimer(f"checkpoint_{step}")
 
     from mirgecom.euler import split_conserved
     cv = split_conserved(discr.dim, q)
     dependent_vars = eos.dependent_vars(cv)
-    expected_state = exact_soln(t=t, x_vec=nodes, eos=eos)
+
+    rank = 0
+    if comm is not None:
+        rank = comm.Get_rank()
+
+    maxerr = 0.0
+    if exact_soln is not None:
+        if profiler is not None:
+            profiler.StartTimer(f"checkpoint_soln_{step}")
+        actx = cv.mass.array_context
+        nodes = thaw(actx, discr.nodes())
+        expected_state = exact_soln(t=t, x_vec=nodes, eos=eos)
+        exp_resid = q - expected_state
+        err_norms = [discr.norm(v, np.inf) for v in exp_resid]
+        maxerr = max(err_norms)
+        if profiler is not None:
+            profiler.EndTimer(f"checkpoint_soln_{step}")
 
     if do_viz:
-        from mirgecom.euler import split_conserved
+        if profiler is not None:
+            profiler.StartTimer(f"checkpoint_viz_{step}")
         io_fields = [
-            ("cv", split_conserved(discr.dim, q)),
-            ("dv", dependent_vars),
-            ("exact_soln", expected_state),
-            ("residual", q - expected_state)
+            ("cv", cv),
+            ("dv", dependent_vars)
         ]
+        if exact_soln is not None:
+            exact_list = [
+                ("exact_soln", expected_state),
+                ("residual", exp_resid)
```

Comment:
I'm kind of skeptical about adding this to the dump. At least paraview can straightforwardly calculate this internally.

Reply:
It was only convenience.  It is gone.